app/tests/test_collections_endpoint.py

```diff
@@ -1,31 +1,65 @@
+import json
 import logging
 from pprint import pformat
 
+from django.apps import apps
 from django.conf import settings
-from django.test import Client
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
 from django.test import TestCase
 
+from rest_framework.authtoken.models import Token
+from rest_framework.test import APIClient
 from rest_framework.test import APIRequestFactory
 
 from stac_api.serializers import CollectionSerializer
 from stac_api.utils import fromisoformat
 
 import tests.database as db
 from tests.utils import get_http_error_description
+from tests.utils import get_sample_data
 from tests.utils import mock_request_from_response
 
 logger = logging.getLogger(__name__)
 
 API_BASE = settings.API_BASE
 
 
+def grant_permissions(model_name, user):
+    '''
+    grant the required permissions to a test user in order to
+    perform e.g. POST or PUT on the specified view
+    :param model_name: name of model, for which the permissions are granted
+    :param user: user, to whom the permissions are granted
+    '''
+    model_dict = apps.all_models['stac_api']
+    # grant permissions needed for specified view
+    for perm in ['add_', 'change_', 'delete_', 'view_']:
+        content_type = ContentType.objects.get_for_model(model_dict[model_name.lower()])
+        permission = Permission.objects.get(
+            codename=perm + model_name.lower(),
+            content_type=content_type,
+        )
+        user.user_permissions.add(permission)
+    user.save()
+    user.refresh_from_db()
+
+
 class CollectionsEndpointTestCase(TestCase):
 
     def setUp(self):
-        self.client = Client()
+        self.client = APIClient()
         self.factory = APIRequestFactory()
         self.collections, self.items, self.assets = db.create_dummy_db_content(4, 4, 4)
         self.maxDiff = None  # pylint: disable=invalid-name
+        self.sample_collections = get_sample_data('collections')
+        self.username = 'SherlockHolmes'
+        self.password = '221B_BakerStreet'
+        self.user = get_user_model().objects.create_user(
```

Comment:
Hmm, if we say the we currently distinguish two types of users, superuser (can do everything, accessing admin, issuing POST/PUT requests etc.) and 'normal'/anonymous users that can just do GET requests I'd probably create a superuser here and omit the explicit permission granting.
Otherwise it will be difficult to track which permissions are exactly required over all and we should rather consider to create an 'apiuser' group or sth like this where we explicitly define the necessary permissions

Reply:
Makes sense. We can still have some "a bit less than superuser"-users later, when needed.

Reply:
I'll add the unit tests for collection POST/PUT endpoint in another PR, soon, after #132 is merged and use superuser and anonymous user for testing.